Hibernate reactive is Java 11 minimum

timyates opened the pull request:
Closes #1317

timyates committed:
reactive hibernate requires java 11

sdelamo commented on starter-core/src/main/java/io/micronaut/starter/feature/database/HibernateReactiveFeatureValidator.java:
```diff
@@ -58,7 +59,11 @@ public void validatePreProcessing(Options options, ApplicationType applicationTy
-        // Do nothing
+        if (features.stream().anyMatch(HibernateReactiveFeature.class::isInstance)
+                && options.getJavaVersion().majorVersion() < JdkVersion.JDK_11.majorVersion()) {
+            System.out.println(options.getJavaVersion().majorVersion());
```
remove println statement

sdelamo commented on starter-core/src/test/groovy/io/micronaut/starter/feature/database/DataHibernateReactiveSpec.groovy:
```diff
@@ -36,9 +37,40 @@ class DataHibernateReactiveSpec extends ApplicationContextSpec {
+                .render()
+
+        then:
+        def ex = thrown(IllegalArgumentException)
```
Much nicer: 

`IllegalArgumentException ex = thrown()`

than: 

`def ex = thrown(IllegalArgumentException)`

sdelamo commented on starter-core/src/test/groovy/io/micronaut/starter/feature/database/HibernateReactiveJpaSpec.groovy:
```diff
@@ -36,10 +37,41 @@ class HibernateReactiveJpaSpec extends ApplicationContextSpec {
+                .render()
+
+        then:
+        def ex = thrown(IllegalArgumentException)
```
Much nicer: 

`IllegalArgumentException ex = thrown()`
        
than: 

`def ex = thrown(IllegalArgumentException)`

sdelamo commented:
I think you have to annotate `DataHibernateReactiveSpec` and `HibernateReactiveJpaSpec` with `@IgnoreIf({ jvm.java8 })`

timyates committed:
Address feedback

timyates commented:
> I think you have to annotate `DataHibernateReactiveSpec` and `HibernateReactiveJpaSpec` with `@IgnoreIf({ jvm.java8 })`

Gah, I had this previously, but it passed locally without it on java 8, so I dropped it...

Added back in 👍

timyates commented:
@sdelamo Any ideas on the failure?  The AmazonCorreto spec is failing

https://ge.micronaut.io/s/zpd3htqttl3xa/tests/overview?outcome=failed

And seems to be failing quite a lot today

https://ge.micronaut.io/scans/failures?failures.failureClassification=all_failures&failures.failureMessage=Failed%20to%20execute%20goal%20org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile%20(default-compile)%20on%20project%20foo:%20Fatal%20error%20compiling&search.timeZoneId=Europe/London

sdelamo commented:
> @sdelamo Any ideas on the failure? The AmazonCorreto spec is failing
> 
> https://ge.micronaut.io/s/zpd3htqttl3xa/tests/overview?outcome=failed
> 
> And seems to be failing quite a lot today
> 
> https://ge.micronaut.io/scans/failures?failures.failureClassification=all_failures&failures.failureMessage=Failed%20to%20execute%20goal%20org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile%20(default-compile)%20on%20project%20foo:%20Fatal%20error%20compiling&search.timeZoneId=Europe/London

Restarted the build and it passed 🤷
